src/go/k8s/pkg/resources/statefulset.go

```diff
@@ -80,19 +81,55 @@ func (r *StatefulSetResource) Ensure(ctx context.Context) error {
 
 	r.LastObservedState = &sts
 
-	// Ensure StatefulSet #replicas equals cluster requirement.
-	if sts.Spec.Replicas != nil && r.pandaCluster.Spec.Replicas != nil && *sts.Spec.Replicas != *r.pandaCluster.Spec.Replicas {
-		r.logger.Info(fmt.Sprintf("StatefulSet %s has replicas set to %d but need %d. Going to update", r.Key().Name, *sts.Spec.Replicas, *r.pandaCluster.Spec.Replicas))
-
-		sts.Spec.Replicas = r.pandaCluster.Spec.Replicas
+	updated := Update(&sts, r.pandaCluster, r.logger)
+	if updated {
 		if err := r.Update(ctx, &sts); err != nil {
-			return fmt.Errorf("failed to update StatefulSet: %w", err)
+			return fmt.Errorf("failed to Update StatefulSet: %w", err)
 		}
 	}
 
 	return nil
 }
 
+// Update ensures StatefulSet #replicas and resources equals cluster requirements.
+func Update(
+	sts *appsv1.StatefulSet,
+	pandaCluster *redpandav1alpha1.Cluster,
+	logger logr.Logger,
+) (updated bool) {
+	return updateReplicasIfNeeded(sts, pandaCluster, logger) || updateResourcesIfNeeded(sts, pandaCluster, logger)
+}
+
+func updateResourcesIfNeeded(
```

Comment:
Not related to this PR:

Can we add the InitContainer resources to the STS? As per https://github.com/vectorizedio/redpanda/pull/602#discussion_r574822596

Reply:
oh yeah, let's do that. Although - should it really be THE SAME resource? I mean to run the init we might not need as much as we would to run a real node, right?

Reply:
As @BenPope stated in https://github.com/vectorizedio/redpanda/pull/602#discussion_r574850329: the thing is scheduler will need to find enough space for both CPU and memory for (redpanda) container. The init container will have at least the same resources available. We should only constraint our init Container to be in Guaranteed QoS not Burstable. 

Reference [Guaranteed QoS class](https://kubernetes.io/docs/tasks/configure-pod-container/quality-service-pod/#create-a-pod-that-gets-assigned-a-qos-class-of-guaranteed)

Reply:
It will be addressed In different PR